### PR TITLE
Buildings open in

### PR DESCRIPTION
**Pull request by mattk410**
Buildings now indicate when they are within 30 minutes of opening, using a Canadian stoplight system (green, yellow, red, yellow)

few-seconds
opening-in

### CODE REVIEW

File: views/building-hours/format-building-hours.js [hunk 1]
```diff
@@ -14,14 +14,18 @@ export function formatBuildingHours(hoursInfo: BuildingInfoType, now: momentT):
...
+  let closeString = close.format(RESULT_FORMAT)
 
   if (openStatus === 'Almost Closed') {
     let closingIn = close.diff(now, 'minutes') + 1
 
     return `Closing in ${closingIn} minutes`
+  } else if (openStatus === 'Almost Open') {
+    let openingIn = open.diff(now, 'minutes') + 1
+
+    return `${openString} / ${closeString} – Opening in ${openingIn} minutes`
```
Comment (hawkrives) [inline views/building-hours/format-building-hours.js]: Didn't we argue against this format for "closing soon"?

File: views/building-hours/is-building-open.js [hunk 2]
```diff
@@ -17,5 +17,8 @@ export function isBuildingOpen(hoursInfo: BuildingInfoType, now: momentT): Build
     }
     return 'Open'
   }
+  if (now.clone().add(30, 'minutes').isAfter(open) && now.isBefore(open)) {
+    return 'Almost Open'
+  }
```
Comment (hawkrives) [inline views/building-hours/is-building-open.js:22]: 👍

**Review by hawkrives — APPROVED** [review]

File: views/building-hours/format-building-hours.js (hunk 1 above)
Comment (drewvolz) [reply views/building-hours/format-building-hours.js]: I didn't catch this on the last PR, but I will mention it here. Will this end up saying "1 minutes" at some point?

Comment (hawkrives) [reply views/building-hours/format-building-hours.js]: Yes. I made a fix for #352 in #360.

**Review by hawkrives — APPROVED** [review]

**Pull request merged**